modin/core/execution/ray/implementations/pandas_on_ray/partitioning/virtual_partition.py

```diff
@@ -0,0 +1,487 @@
+            return df
+
+        drained = super(PandasOnRayDataframeVirtualPartition, self).apply(drain)
+        self.list_of_partitions_to_combine = drained
+        self.call_queue = []
+
+    def add_to_apply_calls(self, func, *args, **kwargs):
+        """
+        Add a function to the call queue.
+
+        Parameters
+        ----------
+        func : callable or ray.ObjectRef
+            Function to be added to the call queue.
+        *args : iterable
+            Additional positional arguments to be passed in `func`.
+        **kwargs : dict
+            Additional keyword arguments to be passed in `func`.
+
+        Returns
+        -------
+        PandasOnRayDataframeVirtualPartition
+            A new ``PandasOnRayDataframeVirtualPartition`` object.
+
+        Notes
+        -----
+        It does not matter if `func` is callable or an ``ray.ObjectRef``. Ray will
+        handle it correctly either way. The keyword arguments are sent as a dictionary.
+        """
+        return type(self)(
+            self.list_of_partitions_to_combine,
+            full_axis=self.full_axis,
+            call_queue=self.call_queue + [(func, args, kwargs)],
+        )
+
+
+class PandasOnRayDataframeColumnPartition(PandasOnRayDataframeVirtualPartition):
+    """
+    The column partition implementation.
+
+    All of the implementation for this class is in the parent class,
+    and this class defines the axis to perform the computation over.
+
+    Parameters
+    ----------
+    list_of_blocks : list
+        List of ``PandasOnRayDataframePartition`` objects.
+    get_ip : bool, default: False
+        Whether to get node IP addresses to conforming partitions or not.
+    full_axis : bool, default: True
+        Whether this partition spans an entire axis of the dataframe.
+    call_queue : list, default: None
+        Call queue that needs to be executed on the partition.
+    """
+
+    axis = 0
+
+
+class PandasOnRayDataframeRowPartition(PandasOnRayDataframeVirtualPartition):
+    """
+    The row partition implementation.
+
+    All of the implementation for this class is in the parent class,
+    and this class defines the axis to perform the computation over.
+
+    Parameters
+    ----------
+    list_of_blocks : list
+        List of ``PandasOnRayDataframePartition`` objects.
+    get_ip : bool, default: False
+        Whether to get node IP addresses to conforming partitions or not.
+    full_axis : bool, default: True
+        Whether this partition spans an entire axis of the dataframe.
+    call_queue : list, default: None
+        Call queue that needs to be executed on the partition.
+    """
+
+    axis = 1
+
+
+@ray.remote
+def deploy_ray_func(func, *args):  # pragma: no cover
+    """
+    Execute a function on an axis partition in a worker process.
+
+    Parameters
+    ----------
+    func : callable
+        Function to be executed on an axis partition.
+    *args : iterable
```

Comment:
Should this be amended to include kwargs?

Reply:
Possibly, but why do we need `kwargs` here?